### PR TITLE
On conflict do nothing for MySql 

### PR DESCRIPTION
**Pull request by darkmmon**
## PR Info

- Partially closes https://github.com/SeaQL/sea-orm/issues/1790

- Dependencies:
  - 

- Dependents:
  - 

## New Features

- [ ]  added do_nothing_mysql([pk_cols]). This is to replace do_nothing() with MySql, require user to input the primary key columns manually.

## Bug Fixes

- [ ]  fixed a bug where do_nothing() will return syntax error with MySql. See New Features for the new function.

## Breaking Changes

- [ ] 

## Changes

- [ ]

### CODE REVIEW

File: tests/mysql/query.rs [hunk 1]
```diff
@@ -1141,6 +1141,32 @@ fn insert_from_select() {
...
+            .on_conflict(
+                OnConflict::new()
+                    .do_nothing_mysql(vec![Glyph::Id])
+                    .to_owned()
+            )
+            .to_string(MysqlQueryBuilder),
+        [
+            r#"INSERT INTO `glyph` (`aspect`, `image`)"#,
+            r#"VALUES ('04108048005887010020060000204E0180400400', 3.1415)"#,
+            r#"ON DUPLICATE KEY UPDATE `id` = VALUES(`id`)"#,
```
Comment (tyt2y3) [inline tests/mysql/query.rs:1164]: Can you make it `id` = `id`?

File: src/query/on_conflict.rs [hunk 2]
```diff
@@ -64,8 +64,84 @@ impl OnConflict {
...
+    /// assert_eq!(
+    ///     query.to_string(MysqlQueryBuilder),
+    ///     [
+    ///         r#"INSERT INTO `glyph` (`aspect`, `image`)"#,
+    ///         r#"VALUES ('abcd', 3.1415)"#,
+    ///         r#"ON DUPLICATE KEY UPDATE `id` = VALUES(`id`)"#,
+    ///     ]
+    ///     .join(" ")
+    /// );
+    pub fn do_nothing_mysql<C, I>(&mut self, pk_cols: I) -> &mut Self
```
Comment (ikrivosheev) [inline src/query/on_conflict.rs:137]: @darkmmon  I'm not sure, that naming is good... 

@tyt2y3 what do you think?

Comment (tyt2y3) [reply src/query/on_conflict.rs:138]: Yes, may be `do_nothing_on` and add `MySQL only` in the doc? https://github.com/SeaQL/sea-query/blob/336bf954b94c618daef0b5c15a0dbf65da12ea25/src/table/column.rs#L384

Comment (ikrivosheev) [reply src/query/on_conflict.rs:138]: I agree with it

File: tests/mysql/query.rs [hunk 3]
```diff
@@ -1161,7 +1183,7 @@ fn insert_on_conflict_0() {
         [
             r#"INSERT INTO `glyph` (`aspect`, `image`)"#,
             r#"VALUES ('04108048005887010020060000204E0180400400', 3.1415)"#,
-            r#"ON DUPLICATE KEY UPDATE `aspect` = VALUES(`aspect`), `image` = VALUES(`image`)"#,
+            r#"ON DUPLICATE KEY UPDATE `aspect` = `aspect`, `image` = `image`"#,
```
Comment (tyt2y3) [inline tests/mysql/query.rs:1186]: Oh sad, this is wrong. `ON DUPLICATE KEY UPDATE id = id` should only applies to `do_nothing_on`.
While the behaviour of `update_columns()` should not be changed.
The two expressions have **different semantics**.

**Review by tyt2y3 — CHANGES_REQUESTED** [review]
As stated

**Pull request closed without merging**